### PR TITLE
TECH-2340 code style checker: add eslint-config-invoca package

### PR DESCRIPTION
**Pull request by rahulk1000**

### CODE REVIEW

**Comment by rahulk1000** [comment]
@caitlinrhd @dtognazzini please code review

**Comment by rahulk1000** [comment]
@caitlinrhd @dtognazzini @alecjacobs5401 code review please

**Comment by rahulk1000** [comment]
Merging this in to make sure it works (end-to-end test). @caitlinrhd @dtognazzini Feel free to still give code review....

File: .hound.yml [hunk 1]
```diff
@@ -2,7 +2,11 @@ ruby:
   config_file: ./ruby/.rubocop.yml
 
 jshint:
-  config_file: ./javascript/.jshint
+  enabled: false
```
Comment (dtognazzini) [inline .hound.yml]: Why this change?

File: javascript/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (dtognazzini) [inline javascript/.eslintrc.json]: The indentation looks off in this file. Mind `COMMAND+OPTION+L`'ing this file to get it formatted?

File: .hound.yml (hunk 1 above)
Comment (rahulk1000) [reply .hound.yml]: I didn't end up running a test. I think it's ok to have jshint explicitly disabled just to be on the safe side...

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
```
[the remark this answers is not in the document]
Comment (rahulk1000) [reply package.json]: I'm not clear on why that is needed. The .eslintrc.json will be included in the "style-guide" node module when we run `npm install` and we will refer to it by doing the following in the web repo's .eslintrc.json file: 

```
{
  "extends": "./node_modules/style-guide/javascript/.eslintrc.json"
}
```

Pretty much what we did in the backdraft repo...

File: javascript/.eslintrc.json (hunk 2 above)
Comment (rahulk1000) [reply javascript/.eslintrc.json]: fixed: https://github.com/Invoca/style-guide/pull/7/commits/447a83b069b8eb932bbce25702830320e16a084e

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (caitlinrhd) [reply package.json]: Looks like it's a matter of what we want to have included in the package:

https://docs.npmjs.com/files/package.json#files
```The optional "files" field is an array of file patterns that describes the entries to be included when your package is installed as a dependency. If the files array is omitted, everything except automatically-excluded files will be included in your publish.```

Are we ok with everything being included?

[the remark this answers is not in the document]
Comment (dtognazzini) [reply package.json]: Ah, I didn't realize that the result of omitting `files` is to _include_ everything. I would have suspected that no `files` means _exclude_ everything. I could see arguments for either. I don't feel strongly either way.

File: .hound.yml (hunk 1 above)
Comment (dtognazzini) [reply .hound.yml]: Ok.
